packages/babel-plugin-transform-react-jsx/src/create-plugin.ts

```diff
@@ -718,7 +727,17 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
           }
 
           if (objs.length === 1) {
-            return objs[0];
+            if (
+              !(
+                t.isSpreadElement(props[0]) &&
+                // If an object expression is spread element's argument
+                // it is very likely to contain __proto__ and we should stop
+                // optimizing spread element
+                t.isObjectExpression(props[0].argument)
```

Comment:
The check here is necessary if we have pattern `{...{ __proto__ }}`, but it is not sufficient: `{...{...{ foo }}}`, after optimized into `{...{ foo }}` by `accumulateAttribute`, won't further pass the check here. I think this is fine as practically such pattern is rare.